plugins/costinsight/src/hooks/utility.js

```diff
@@ -0,0 +1,146 @@
+                        recurse(schemaCur.fields[p], rowsCur, colName + "." + schemaCur.fields[p].name)
+                    } else {
+                        recurse(schemaCur.fields[p], rowsCur.v.f[valIndex], colName + "." + schemaCur.fields[p].name)
+                    }
+
+                    valIndex++
+                }
+            }
+
+            if (schemaCur.mode === "REPEATED") {
+                result[colName] = []
+                for (var x in rowsCur.v) {
+                    recurse(schemaCur.fields, rowsCur.v[x], colName)
+                }
+            }
+        } else {
+            if (schemaCur.mode === "REPEATED") {
+                if (rowsCur.v !== null) {
+                    result[colName] = rowsCur.v.map((value, index) => { return value.v })
+                } else {
+                    result[colName] = [null]
+                }
+
+            } else if (Array.isArray(result[colName])) {
+                let nextRow = {}
+                for (var j in schemaCur) {
+                    nextRow[colName + "." + schemaCur[j].name] = Array.isArray(rowsCur.v.f[j].v) ? rowsCur.v.f[j].v.map((value, index) => { return value.v }) : rowsCur.v.f[j].v
+                }
+                result[colName].push(nextRow)
+            } else {
+                if (colName !== "")
+                    result[colName] = rowsCur.v
+            }
+        }
+    }
+
+    for (var r = 0, rowsCount = rows.length; r < rowsCount; r++) {
+        var result = {};
+        recurse(schema, rows[r], "")
+        resultRows.push(result)
+    }
+    return resultRows
+};
+/**
+ * process response end*/
+
+/**
+ * Object Data converstion utility method 
+ * @param {*} input must be Object
+ * @returns 
+ */
+
+export const withoutKeyObject = (input) => {
+    let result = {};
+    for (const key in input) {
+        if (!input.hasOwnProperty(key)) {
+            continue;
+        }
+        if (typeof input[key] === "object" && !Array.isArray(input[key])) {
+            var subFlatObject = withoutKeyObject(input[key]);
+            for (const subkey in subFlatObject) {
+                result[key + "_" + subkey] = subFlatObject[subkey];
+            }
+        } else {
+            result[key] = input[key];
+        }
+    }
+    return result;
+}
+
+/**
+ * Object Data converstion utility method!
+ * @param {*} input must be data Object
+ * @param {*} keyName must be the specific key which we need to retrieve the values
+ * @returns 
+ */
+export const withKeyObject = (input, keyName) => {
+    var result = {};
+    for (const key in input) {
+        const newKey = keyName ? `${keyName}_${key}` : key;
+        if (typeof input[key] === "object" && !Array.isArray(input[key])) {
+            result = { ...result, ...withKeyObject(input[key], newKey) }
+        } else {
+            result[newKey] = input[key];
+        }
+    }
+    return result;
+};
+
+/**
+ * Chart Style Properties
+ */
+export const useStyles = makeStyles((theme) => ({
```

Comment:
Could we abstract the styles to a .css file?

Reply:
No comments for now.